packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -147,19 +151,68 @@ class SegmentedControl<T> extends StatefulWidget {
   _SegmentedControlState<T> createState() => _SegmentedControlState<T>();
 }
 
-class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
+class _SegmentedControlState<T> extends State<SegmentedControl<T>>
+    with TickerProviderStateMixin<SegmentedControl<T>> {
   T _pressedKey;
 
-  void _onTapDown(T currentKey) {
-    setState(() {
-      _pressedKey = currentKey;
-    });
+  final List<AnimationController> _selectionControllers = <AnimationController>[];
+  final List<ColorTween> _childTweens = <ColorTween>[];
+
+  static final ColorTween forwardBackgroundColorTween = new ColorTween(
+    begin: _kPressedBackground,
+    end: CupertinoColors.activeBlue,
+  );
+
+  static final ColorTween reverseBackgroundColorTween = new ColorTween(
+    begin: CupertinoColors.white,
+    end: CupertinoColors.activeBlue,
+  );
+
+  static final ColorTween textColorTween = new ColorTween(
+    begin: CupertinoColors.activeBlue,
+    end: CupertinoColors.white,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    for (T key in widget.children.keys) {
+      final AnimationController animationController = createAnimationController();
+      if (widget.groupValue == key) {
+        _childTweens.add(reverseBackgroundColorTween);
+        animationController.forward();
```

Comment:
if I read this right, this means that any time you enter a page that has a segmented control with an already-selected value, it'll fade in, right? I think for the initial value we want things to be static.

Reply:
@Hixie That makes sense - I had it this way because the first action on the animation controller for the initially selected element will be a reverse, which you can't do if the animation hasn't gone forward already. I could potentially make a new Tween that has the begin/end colors switched - but I think there must be a more intuitive solution?